### PR TITLE
add alignmentSieve

### LINKED ISSUE #6690: new module: DEEPTOOLS/ALIGNMENTSIEVE
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Further Information

Added as a local module to nf-core/atacseq [here](https://github.com/nf-core/atacseq/pull/375)

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by lpantano**
## PR checklist

Add new sub-module to add feature in atacseq https://github.com/nf-core/atacseq/pull/375

Closes #6690 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by lpantano** [comment]
@JoseEspinosa , I saw the issue you created and already started, there is at least one thing missing. The snapshot is not stable so the nf-test should test something else, and not the md5...  Do you know alternatives?

**Comment by lpantano** [comment]
I think is ready, the errors there I believe is due to something else right not.

```
  remote: git: 'lariat' is not a git command. See 'git --help'.        
```

File: modules/nf-core/deeptools/alignmentsieve/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.logs,
+                                  process.out.versions)
+                                  .match()
```
Comment (SPPearce) [inline modules/nf-core/deeptools/alignmentsieve/tests/main.nf.test:31]: Try this for the bam file, it uses the [nft-bam plugin ](https://nvnieuwk.github.io/nft-bam/latest/):
```suggestion
                { assert snapshot(process.out.logs,
                                  bam(process.out.bam[0][1]).getReadsMD5(),
                                  process.out.versions)
                                  .match()
```

File: modules/nf-core/deeptools/alignmentsieve/meta.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          pattern: "versions.yml"
+authors:
+  - "@FriederikeHanssen"
+  - "@SusiJo"
+  - "@JoseEspinosa"
+  - "@lpantano"
+maintainers:
+  - "@FriederikeHanssen"
+  - "@SusiJo"
+  - "@JoseEspinosa"
```
Comment (JoseEspinosa) [inline modules/nf-core/deeptools/alignmentsieve/meta.yml:60]:
```suggestion
  - "@lpantano"
```

File: modules/nf-core/deeptools/alignmentsieve/meta.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+name: deeptools_alignmentSieve
+description: This tool filters alignments in a BAM/CRAM file according the the specified parameters.
+keywords:
+  - filter
```
Comment (JoseEspinosa) [inline modules/nf-core/deeptools/alignmentsieve/meta.yml:4]:
```suggestion
  - ATACseq
  - filter
```

**Review by JoseEspinosa — COMMENTED** [review]
Nice @lpantano! I just dropped some small suggestions.
Seems that the tests are not fixed yet though 😢

**Review by JoseEspinosa — COMMENTED** [review]
Maybe try to delete the old snapshots and regenerate them and tests might work.

**Comment by lpantano** [comment]
agreed, there were some commits to use specific functions for BAM md5 but the snapshots are the old ones. I will update. Thanks!

**Comment by lpantano** [comment]
I think now it is good, but not sure why is waiting for status to be reported.

**Review by JoseEspinosa — APPROVED** [review]
LGTM! 
I merged with master let's see if tests run now

**Pull request merged**